src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/DocumentState.cs

```diff
@@ -298,18 +318,15 @@ public Task<(RazorCodeDocument, VersionStamp, VersionStamp)> GetGeneratedOutputA
                     throw new ArgumentNullException(nameof(document));
                 }
 
-                if (TaskUnsafe == null)
```

Comment:
Since `TaskUnsafe` can now transition to `null` I had to remove the optimised non-locked path here to account for that.